### PR TITLE
Try to prevent additional issues with writing the changelogs to file

### PR DESCRIPTION
**Pull request by zackkrida**
I am _not_ a shell script expert but am curious if this might fix issues I'm still seeing with drafting certain release notes. See the [recent run here](https://github.com/WordPress/openverse/actions/runs/4735022950/jobs/8404730500) which errored due to a "syntax error near unexpected token `(".

### CODE REVIEW

**Comment by AetherUnbound** [comment]
It looks like that run failed because a `'` character within the changelog body closed the changelog string before the final `'` was supposed to. Unfortunately, I don't think this change will address the problem either because the double quotes in `"${{ steps.release-drafter.outputs.body }}"` puts us back in the same position we were in #1193 where backticks were being interpolated. I think we might need to use a github expression to replace or escape the `'` characters in the changelog as part of `${{ steps.release-drafter.outputs.body }}`.

**Review by dhruvkb — COMMENTED** [review]
This is not directly helpful but I'd recommend using Python for anything more complicated than a simple `echo`. That way the expression can be interpolated into a triple quotes raw string, escaping everything.

```python
value = r"""${{ text }}"""
```

**Review by AetherUnbound — APPROVED** [review]
LGTM! 🤞🏼

**Comment by zackkrida** [comment]
I'm going to try a frontend test run with this PR.

**Comment by zackkrida** [comment]
@AetherUnbound @dhruvkb I made a test run of this but I think the path was incorrect:

https://github.com/WordPress/openverse/actions/runs/4736717050/jobs/8408615526

```
Traceback (most recent call last):
  File "/home/runner/work/openverse/openverse/automations/python/workflows/write_changelog.py", line 10, in <module>
    with open(file_path, "w") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'documentation/changelogs/frontend/2023.04.18.20.33.52.md'
Error: Process completed with exit code 1.
```

I think it might relate to the working directory in the workflow configuration being         `working-directory: automations/python/workflows` and the relative path in the script being `file_path = f"documentation/changelogs/{app}/{tag_date}.md"`.

Perhaps I shouldn't set the working directory and instead invoke ` python  automations/python/workflows/write_changelog.py` in the run command 🤔

**Comment by AetherUnbound** [comment]
Oh shoot, that's totally it. Adding a suggestion for the script now regarding this.

File: automations/python/workflows/write_changelog.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+app = os.environ["APP"]
+tag_date = os.environ["DATE"]
+release_body = os.environ["RELEASE_BODY"]
+
+file_path = f"documentation/changelogs/{app}/{tag_date}.md"
+
+with open(file_path, "w") as f:
+    f.write(f"# {tag_date}\n\n")
+    f.write(f"""{release_body}\n""")
```
Comment (AetherUnbound) [inline automations/python/workflows/write_changelog.py:12]: Potential change to make this resilient to working directory changes

```suggestion
import os
from pathlib import Path


app = os.environ["APP"]
tag_date = os.environ["DATE"]
release_body = os.environ["RELEASE_BODY"]

file_path = Path(__file__).parents[3] / f"documentation/changelogs/{app}/{tag_date}.md"

with file_path.open("w") as f:
    f.write(f"# {tag_date}\n\n")
    f.write(f"""{release_body}\n""")
```

Comment (sarayourfriend) [reply automations/python/workflows/write_changelog.py:12]: FWIW with Pathlib you don't even need to use `open`:

```py
file_path.write_text(
    """
    # {tag_date}

    {release_body}
    """
)
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! Good call from @dhruvkb to use Python instead.

File: automations/python/workflows/write_changelog.py (hunk 1 above)
Comment (AetherUnbound) [reply automations/python/workflows/write_changelog.py:12]: Good point 🙂

**Pull request merged**